recipes/aws-cdi-sdk/all/patches/001-Makefile.patch

```diff
@@ -0,0 +1,277 @@
+ ## library build definitions
+ # "top level" subtrees
+ top.cdi := $(top.src)/cdi
+@@ -120,22 +100,18 @@ src_dir.test_unit := $(top.test_unit)
+ # directories with header files
+ include_dir.cdi := $(top.cdi)
+ include_dirs.cdi := $(foreach dir,sdk cdi common,$(include_dir.$(dir)))
+-include_dirs.libfabric := $(top.libfabric)/include $(build_dir.libfabric)
+ 
+ # generate various lists for building SDK library
+ srcs.cdi := $(foreach ext,c cpp,$(wildcard $(src_dir.cdi)/*.$(ext)))
+ srcs.cdi += queue.c fifo.c list.c logger.c os_linux.c pool.c
+ objs.cdi := $(addprefix $(build_dir.obj)/,$(patsubst %.cpp,%.o,$(patsubst %.c,%.o,$(notdir $(srcs.cdi)))))
+ headers.cdi := $(foreach dir,$(include_dirs.cdi),$(wildcard $(dir)/*.h))
+-include_opts.cdi := $(foreach proj,cdi libfabric,$(addprefix -I,$(include_dirs.$(proj))))
++include_opts.cdi := $(foreach proj,cdi,$(addprefix -I,$(include_dirs.$(proj))))
+ depends.cdi := $(patsubst %.o,%.d,$(objs.cdi))
+ 
+ # the end goal of building the SDK library
+ libsdk := $(build_dir.lib)/libcdisdk.so.2.0
+ 
+-# the end goal of building the libfabric shared library
+-libfabric := $(build_dir.lib)/libfabric.so.1
+-
+ ## test program definitions
+ # test specific source files are all in one directory
+ src_dir.test := $(top.src)/test
+@@ -186,10 +162,10 @@ headers.all := $(foreach dir,cdi test test_common test_min_tx test_min_rx test_u
+ # augment compiler flags
+ COMMON_COMPILER_FLAG_ADDITIONS := \
+ 	$(include_opts.cdi) $(include_opts.test) \
+-	$(EXTRA_COMPILER_FLAG_ADDITIONS) \
+-	-Wall -Wextra -Werror -pthread -fPIC \
++	-Wall -Wextra -pthread -fPIC \
+ 	-D_LINUX -D_POSIX_C_SOURCE=200112L \
+-	$(sanitize_opts)
++	$(sanitize_opts) \
++	$(EXTRA_COMPILER_FLAG_ADDITIONS)
+ ifeq ($(config), debug)
+     COMMON_COMPILER_FLAG_ADDITIONS += -O0 -g -DDEBUG
+ else
+@@ -203,7 +179,7 @@ CXXFLAGS += $(COMMON_COMPILER_FLAG_ADDITIONS) --std=c++11
+ # The only libraries needed here are those that present new dependencies beyond what libcdisdk.so already requires.
+ # An rpath is specified so cdi_test can find libcdisdk.so.2 in the same directory as cdi_test or in a sibling directory
+ # named lib.
+-CDI_LDFLAGS = $(LDFLAGS) -L$(build_dir.lib) -lcdisdk -lfabric $(EXTRA_LD_LIBS) -lncurses -lm $(aws_sdk_library_flags) \
++CDI_LDFLAGS = $(LDFLAGS) -L$(build_dir.lib) -lcdisdk -lfabric $(EXTRA_LD_LIBS) -lncurses -lm \
+ 	   -Wl,-rpath,\$$ORIGIN:\$$ORIGIN/../lib64:\$$ORIGIN/../lib
+ 
+ # docs go into the build directory but are not specific to release/debug
+@@ -222,37 +198,11 @@ endif
+ 
+ # default build target
+ .PHONY : all
+-all : libfabric libsdk test docs docs_api $(EXTRA_ALL_TARGETS)
++all : libsdk test docs docs_api $(EXTRA_ALL_TARGETS)
+ 
+ # Build only the libraries
+ .PHONY : lib
+-lib : libfabric libsdk
+-
+-# Ensure that the location of the AWS SDK source code tree was specified unless explicitly opted out. Define and augment
+-# some variables needed for building and linking to the AWS SDK.
+-ifeq ($(require_aws_sdk),yes)
```

Comment:
This does not seem required to build the library 🤔 can you explain what this patch is for please?

Reply:
the original package builds libfabric and the AWS SDK itself and here I remove that part